builtin/credential/approle/path_role.go

```diff
@@ -1926,39 +1925,12 @@ func (b *backend) handleRoleSecretIDCommon(req *logical.Request, data *framework
 	}, nil
 }
 
-// roleIDLock is used to get a lock from the pre-initialized map
-// of locks. Map is indexed based on the first 2 characters of the
-// RoleID, which is a random UUID. If the input is not hex encoded
-// or if it is empty a "custom" lock will be returned.
-func (b *backend) roleIDLock(roleID string) *sync.RWMutex {
-	var lock *sync.RWMutex
-	var ok bool
-	if len(roleID) >= 2 {
-		lock, ok = b.roleIDLocksMap[roleID[0:2]]
-	}
-	if !ok || lock == nil {
-		lock = b.roleIDLocksMap["custom"]
-	}
-	return lock
+func (b *backend) roleIDLock(roleID string) *locksutil.LockEntry {
```

Comment:
Up to you. They're one liners but still less boilerplate than copying that line everywhere else.